test/fixtures/curve/basepool.vy

```diff
@@ -507,6 +507,103 @@ def _calc_withdraw_one_coin(_token_amount: uint256, i: uint256) -> (uint256, uin
 def calc_withdraw_one_coin(_token_amount: uint256, i: uint256) -> uint256:
     return self._calc_withdraw_one_coin(_token_amount, i)[0]
 
+@external
+@nonreentrant('lock')
+def remove_liquidity(_burn_amount: uint256, _min_amounts: uint256[N_COINS]):
+    """
+    This withdrawal method is very safe, does no complex math
+    """
+    total_supply: uint256 = self.token.totalSupply()
+    amounts: uint256[N_COINS] = empty(uint256[N_COINS])
+    amount: uint256 =  _burn_amount - 1 # favor LPs a little bit
+
+    for i in range(N_COINS):
+        old_balance: uint256 = self.balances[i]
+        value: uint256 = old_balance * amount / total_supply
+        assert value >= _min_amounts[i], "Withdrawal resulted in fewer coins than expected"
+        self.balances[i] = old_balance - value
+        amounts[i] = value
+        # sim: comment-out the interaction with coin
+        # ----------------------------------------------------------
+        # response: Bytes[32] = raw_call(
+        #     self.coins[i],
+        #     concat(
+        #         method_id("transfer(address,uint256)"),
+        #         convert(_receiver, bytes32),
+        #         convert(value, bytes32),
+        #     ),
+        #     max_outsize=32,
+        # )
+        # if len(response) > 0:
+        #     assert convert(response, bool)
+
+    total_supply -= _burn_amount
+    self.token.burnFrom(msg.sender, _burn_amount)  # dev: insufficient funds
+
+    log RemoveLiquidity(msg.sender, amounts, empty(uint256[N_COINS]), total_supply)
+
+@external
+@nonreentrant('lock')
+def remove_liquidity_imbalance(
+    _amounts: uint256[N_COINS],
+    _max_burn_amount: uint256,
+) -> uint256:
+    """
+    Remove an imbalanced amount of coins from the pool.
+    """
+    amp: uint256 = self._A()
+    rates: uint256[N_COINS] = RATES
+    old_balances: uint256[N_COINS] = self.balances
+    D0: uint256 = self.get_D_mem(old_balances, amp)
+
+    new_balances: uint256[N_COINS] = old_balances
+    for i in range(N_COINS):
```

Comment:
- The commented-out coin transfer logic should be moved to a commented-out for loop at the end of the function. The onchain 3pool does it that way: https://etherscan.io/address/0xbebc44782c7db0a1a60cb6fe97d0b483032ff1c7#code (see Lines 541-542 and 563-577 in link for example)
- The `assert` statement string could be more helpful
- The `print` statements can be removed

Reply:
Sorry, I was using the print and assert statements to fix a testing error and forgot to remove those